python/src/aiconfig/util/config_utils.py

```diff
@@ -10,10 +11,28 @@
     from ..schema import AIConfig
 
 
-def get_api_key_from_environment(api_key_name: str):
-    if api_key_name not in os.environ:
-        raise Exception("Missing API key '{}' in environment".format(api_key_name))
+def maybe_get_api_key_from_environment(
```

Comment:
nit, generally not a fan of using 'maybe' in function names when it can be derived from type. And here it is a bit confusing because by default it actually has to either return a value or throw -- there's no 'maybe' case until you specify required false

Reply:
agreed

Reply:
Spoke to @rossdanlm . We're going to do optional here and a follow-up PR to switch to Result, because None is actually a valid value.